lib/internal/worker.js

```diff
@@ -104,7 +104,7 @@ class Worker extends EventEmitter {
       filename = path.resolve(filename);
 
       const ext = path.extname(filename);
-      if (ext !== '.js' && ext !== '.mjs') {
+      if (!/^\.[cm]?js$/.test(ext)) {
```

Comment:
Nit: why not just add another `&& ext !== '.cjs'`, that would be both simpler to understand and faster to run. I don't see the benefits of using regexp here.